OpenSim/Simulation/Model/Geometry.h

```diff
@@ -611,7 +615,12 @@ class OSIMSIMULATION_API Mesh : public Geometry
     /// Method to map Mesh to Array of SimTK::DecorativeGeometry.
     void implementCreateDecorativeGeometry(
         SimTK::Array_<SimTK::DecorativeGeometry>& decoGeoms) const override;
-
+private:
+    // We cache the DecorativeMeshFile and a flag indicating if we tried to
+    // load the mesh from file so we don't try loading from disk every frame.
+    // These are mutable since they're not part of the public interface.
+    mutable std::shared_ptr<SimTK::DecorativeMeshFile> cachedMesh;
+    mutable bool triedToLoad;
```

Comment:
Would it be possible to use the pointer `cachedMesh` as the flag (and so remove this member variable)? The smart pointers (both unique and shared) have an operator bool to tell you if they are managing an object:

![image](https://cloud.githubusercontent.com/assets/846001/14308136/053005d6-fb89-11e5-9347-210464ad2306.png)
http://en.cppreference.com/w/cpp/memory/unique_ptr

So you might be able to replace `if (!triedToLoad)` with `if (!cachedMesh)`.